src/sentence.rs

```diff
@@ -0,0 +1,373 @@
+
+                let next_cat = se::sentence_category(next_char);
+
+                self.pos += next_char.len_utf8();
+                self.state = self.state.next(next_cat);
+
+                match next_cat {
+                    // SB1 https://unicode.org/reports/tr29/#SB1
+                    _ if state_before.match1(StatePart::Sot) =>
+                        return Some(position_before),
+
+                    // SB2 is handled when inner iterator (chars) is finished
+
+                    // SB3 https://unicode.org/reports/tr29/#SB3
+                    SentenceCat::SC_LF if state_before.match1(StatePart::CR) =>
+                        continue,
+
+                    // SB4 https://unicode.org/reports/tr29/#SB4
+                    _ if state_before.match1(StatePart::Sep)
+                        || state_before.match1(StatePart::CR)
+                        || state_before.match1(StatePart::LF)
+                    => return Some(position_before),
+
+                    // SB5 https://unicode.org/reports/tr29/#SB5
+                    SentenceCat::SC_Extend |
+                    SentenceCat::SC_Format => self.state = state_before,
+
+                    // SB6 https://unicode.org/reports/tr29/#SB6
+                    SentenceCat::SC_Numeric if state_before.match1(StatePart::ATerm) =>
+                        continue,
+
+                    // SB7 https://unicode.org/reports/tr29/#SB7
+                    SentenceCat::SC_Upper if state_before.match2(StatePart::UpperLower, StatePart::ATerm) =>
+                        continue,
+
+                    // SB8 https://unicode.org/reports/tr29/#SB8
+                    _ if match_sb8(&state_before, &self.string[position_before..]) =>
+                        continue,
+
+                    // SB8a https://unicode.org/reports/tr29/#SB8a
+                    SentenceCat::SC_SContinue |
+                    SentenceCat::SC_STerm |
+                    SentenceCat::SC_ATerm if match_sb8a(&state_before) =>
+                        continue,
+
+                    // SB9 https://unicode.org/reports/tr29/#SB9
+                    SentenceCat::SC_Close |
+                    SentenceCat::SC_Sp |
+                    SentenceCat::SC_Sep |
+                    SentenceCat::SC_CR |
+                    SentenceCat::SC_LF if match_sb9(&state_before) =>
+                        continue,
+
+                    // SB10 https://unicode.org/reports/tr29/#SB10
+                    SentenceCat::SC_Sp |
+                    SentenceCat::SC_Sep |
+                    SentenceCat::SC_CR |
+                    SentenceCat::SC_LF if match_sb8a(&state_before) =>
+                        continue,
+
+                    // SB11 https://unicode.org/reports/tr29/#SB11
+                    _ if match_sb11(&state_before) =>
+                        return Some(position_before),
+
+                    // SB998 https://unicode.org/reports/tr29/#SB998
+                    _ => continue
```

Comment:
Tried to double check these rules with respect to the UAX#28 spec, can't see any inconsistencies. Also, I imagine the included test would break if something wasn't right.